src/ec2_metadata/__init__.py

```diff
@@ -97,6 +97,19 @@ def iam_info(self) -> Optional[Dict[str, Any]]:
             return None
         return resp.json()
 
+    @cached_property
```

Comment:
these can't be cached, because they are temporary?

Reply:
Will fix.